### PR TITLE
ENGDESK-17880: Skip audio validation when audio port is set to 0

### PR DESCRIPTION
**Pull request by dev-ryanc**

### CODE REVIEW

File: src/switch_core_media.c [hunk 1]
```diff
@@ -5523,6 +5525,11 @@ SWITCH_DECLARE(uint8_t) switch_core_media_validate_common_audio_sdp(switch_core_
 		}
 	}
 
+	if (no_audio) {
+		switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "No Audio found! Skip audio validation.\n");
+		return -1;
```
Comment (piotrgregor) [inline src/switch_core_media.c:5530]: Return type is unsigned, this will overflow to 255.

**Review by piotrgregor — CHANGES_REQUESTED** [review]

**Review by piotrgregor — CHANGES_REQUESTED** [review]
Audio validation should not be skipped for 0 port. Port 0 is valid port number that indicates stream rejection (ENGDESK-18035 for details). Seems that this would be only a work around more severe problem that I'm already checking under ENGDESK-18035

**Comment by dev-ryanc** [comment]
> Audio validation should not be skipped for 0 port. Port 0 is valid port number that indicates stream rejection (ENGDESK-18035 for details). Seems that this would be only a work around more severe problem that I'm already checking under ENGDESK-18035

see [slack thread](https://telnyx.slack.com/archives/C805L26HY/p1661162805836429?thread_ts=1660831440.495329&cid=C805L26HY)

File: src/switch_core_media.c (hunk 1 above)
Comment (dev-ryanc) [reply src/switch_core_media.c:5530]: Good catch, fix it

**Review by StefanYohansson — APPROVED** [review]

**Pull request merged**